lib/mix/lib/mix/cli.ex

```diff
@@ -9,6 +9,7 @@ defmodule Mix.CLI do
     Mix.Local.append_paths
 
     if System.get_env("MIX_QUIET"), do: Mix.shell(Mix.Shell.Quiet)
+    if System.get_env("MIX_DEBUG"), do: Mix.debug(true)
```

Comment:
In the man page at least. We could probably add an environment variables section to the `Mix` module docs too. Could you please act on both? :)